packages/client/src/pages/quiz-master-session/ui/QuizMasterHeader.tsx

```diff
@@ -1,37 +1,55 @@
-import { usePersistState } from '@/shared/hooks/usePersistState';
+import { Socket } from 'socket.io-client';
 import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 
-import { Socket } from 'socket.io-client';
+import { usePersistState } from '@/shared/hooks/usePersistState';
+import { useQuizContext } from '../hooks/useQuizContext';
+import { getCookie } from '@/shared/utils/cookie';
+import { clearLocalStorage } from '@/shared/utils/clearLocalStorage';
+import { MASTER_LOCAL_STORAGE_KEYS } from '@/shared/constants/masterLocalStorageKey';
 
 interface QuizMasterHeaderProps {
-  quizData: QuizData;
-  startTime: number;
-  timeLimit: number;
-  handleNextQuiz: () => void;
   pinCode: string;
+  id: string;
   socket: Socket;
 }
 
-export default function QuizMasterHeader({
-  quizData,
-  startTime,
-  timeLimit,
-  handleNextQuiz,
-  pinCode,
-  socket,
-}: QuizMasterHeaderProps) {
-  const [remainingTime, setRemainingTime] = usePersistState('remainingTime', timeLimit);
+export default function QuizMasterHeader({ pinCode, socket, id }: QuizMasterHeaderProps) {
+  const navigate = useNavigate();
+  const { quiz, refetch, setInitializeStates } = useQuizContext();
+
+  const [remainingTime, setRemainingTime] = usePersistState(
+    'remainingTime',
+    quiz.currentQuizData.timeLimit,
+  );
+
+  const handleNextQuiz = () => {
+    if (quiz.isLast) {
+      socket.emit('end quiz', { pinCode, sid: getCookie('sid') });
+      clearLocalStorage(MASTER_LOCAL_STORAGE_KEYS);
+      navigate(`/quiz/session/${pinCode}/end`);
+      return;
+    }
+
+    socket.emitWithAck('start quiz', { pinCode, sid: getCookie('sid') }).then(() => {
+      clearLocalStorage(MASTER_LOCAL_STORAGE_KEYS);
+      navigate(`/quiz/session/host/${pinCode}/${parseInt(id as string) + 1}`);
```

Comment:
```suggestion
      navigate(`/quiz/session/host/${pinCode}/${parseInt(id) + 1}`);
```

`id`의 타입이 string인데 타입 단언을 안해도 괜찮지 않나요?